src/components/screens/layout-screen-html.tsx

```diff
@@ -0,0 +1,233 @@
+import styles from "../../styles/components/layout-screen/layout-editor-view";
+import {
+  CircularProgress,
+  TextField,
+  Select,
+  MenuItem,
+  InputLabel,
+  FormControl,
+  SelectChangeEvent,
+} from "@mui/material";
+import { WithStyles } from '@mui/styles';
+import withStyles from '@mui/styles/withStyles';
+import { KeycloakInstance } from "keycloak-js";
+import { PageLayout, Exhibition, DeviceModel, ScreenOrientation, SubLayout } from "../../generated/client";
+import BasicLayout from "../layouts/basic-layout";
+import ElementNavigationPane from "../layouts/element-navigation-pane";
+import EditorView from "../editor/editor-view";
+import { AccessToken, ActionButton } from '../../types';
+import strings from "../../localization/strings";
+import theme from "../../styles/theme";
+
+/**
+ * Component props
+ */
+interface Props extends WithStyles<typeof styles> {
+  history: History;
+  keycloak: KeycloakInstance;
+  accessToken: AccessToken;
+  exhibitions: Exhibition[];
+  deviceModels: DeviceModel[];
+  layoutId: string;
+  layout?: PageLayout;
+  layouts: PageLayout[];
+  subLayouts: SubLayout[];
+  setLayouts: typeof setLayouts;
+  setSelectedLayout: typeof setSelectedLayout;
+}
+
+/**
+ * Component for html layout screen
+ */
+export const LayoutScreenHTML: React.FC<Props> = ({
+  history,
+  keycloak,
+  deviceModels,
+  layout,
+  classes
+}) => {
+  const [ name, setName ] = useState("");
+  const [ deviceModelId, setDeviceModelId ] = useState("");
+  const [ screenOrientation, setScreenOrientation ] = useState<ScreenOrientation>(ScreenOrientation.Portrait);
+  const [ view, setView ] = useState("VISUAL")
+  const [ dataChanged, setDataChanged ] = useState(false);
+
+  /**
+   * Event handler for name input change
+   *
+   * @param event event
+   */
+  const onNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setName(event.target.value);
+    setDataChanged(true);
+  }
+
+  /**
+   * Event handler for screen orientation select change
+   *
+   * @param event event
+   */
+  const onScreenOrientationChange = (event: SelectChangeEvent<ScreenOrientation>) => {
+    setScreenOrientation(event.target.value as ScreenOrientation);
+    setDataChanged(true);
+  }
+
+  /**
+   * Event handler for device model select change
+   *
+   * @param event event
+   */
+  const onDeviceModelChange = (event: SelectChangeEvent<string>) => {
+    setDeviceModelId(event.target.value as string);
+    setDataChanged(true);
+  }
+
+  /**
+   * Gets action buttons
+   *
+   * @returns action buttons as array
+   */
+  const getActionButtons = (): ActionButton[] => (
+    [
+      {
+        name: view === "CODE" ?
+          strings.exhibitionLayouts.editView.switchToVisualButton :
+          strings.exhibitionLayouts.editView.switchToCodeButton,
+        action: () => view === "CODE" ? setView("VISUAL") : setView("CODE"),
+      },
+      {
+        name: strings.exhibitionLayouts.editView.saveButton,
```

Comment:
Save functionality is in place in the API so therefore this action could just save it.